Add missing Redshift driver password field

tlrobinson opened the pull request:
Resolves #8267

tlrobinson committed:
Add missing Redshift driver password field

senior commented on src/metabase/driver/redshift.clj:
```diff
@@ -90,6 +90,7 @@
                                                      :placeholder  (tru "toucan_sightings"))
                                                    driver/default-user-details
+                                                   driver/default-password-details
                                                    driver/default-port-details]))
```
Where I messed up is adding `driver/default-port-details` rather than `driver/default-password-details`. I'll push up another commit that removes the second port entry.

iethree committed:
Remove the redundant Redshift port config

The redshift connection configuration details got messed up in the
refactor. Instead of the password field a second port field got
included. The password field was put back in but the second port field
needed to get removed.
